common/unsafe/src/main/java/org/apache/spark/unsafe/types/UTF8String.java

```diff
@@ -835,6 +835,12 @@ public static UTF8String concatWs(UTF8String separator, UTF8String... inputs) {
     return res;
   }
 
+  public UTF8String replace(UTF8String search, UTF8String replace) {
+    String replaced = toString().replace(
+      search.toString(), replace.toString());
```

Comment:
When the search string is an empty string, we should not change the original string.